lua/iron/init.lua

```diff
@@ -374,30 +374,28 @@ iron.core.send_chunk = function(mode, mtype)
 
   iron.core.send(ft, lines)
 
-  local mark = vim.api.nvim_buf_get_extmark_by_id(0, iron.namespace, iron.mark.save_pos)
+  local mark = vim.api.nvim_buf_get_extmark_by_id(0, iron.namespace, iron.mark.save_pos, {})
 
   if #mark ~= 0 then
     -- winrestview is 1-based
     vim.fn.winrestview({lnum = mark[1] + 1, col = mark[2] + 1})
-    vim.api.nvim_buf_del_extmark(0, iron.namespace, 20)
+    vim.api.nvim_buf_del_extmark(0, iron.namespace, iron.mark.save_pos)
   end
 
   vim.api.nvim_buf_set_extmark(
     0,
     iron.namespace,
-    iron.mark.begin_last,
-    b_line - 1,
+    b_line - 2,
```

Comment:
Hey @GCBallesteros! My apologies for coming out of nowhere. I was also testing this on the latest pre-release and found maybe a third of the api changes that you did, thanks for putting these together!

Not sure how much this helps, but when testing this out, the only nit I've found is that it seems to kind of break the flow of:

- Go to top of file
- Highlight whole file
- Copy all contents to the REPL

My guess, is that this is failing because the selection is expecting an additional space above the mark. The error it seems to throw is "line value outside range".

Changing this back to `b_line - 1` seems to get rid of that situation.

Out of curiosity, what situations was it starting a line too late? Not sure if what I'm talking about is intended behavior or just coincidence. 😄

Reply:
Nice to meet you! I found this problem when using the repeat command functionality, the default `cp` mapping. It should send to the REPL the last sent region. E.g. if you do `ctr2j` you will send the line on the cursor +2. Without the change you comment, `cp` will not send the first of the lines. 
![bug_iron](https://user-images.githubusercontent.com/6859287/94897957-9661fe80-0488-11eb-88cb-eb4acbf0e034.gif)

I have now noticed that this only happens with the normal mode mapping. The visual mode mapping will, with my pull request, pick up an additional unintended line. That explains the breaking of the flow you describe. If you select visually the top row of the buffer then it tries to send an additional out of bounds one.

So the change fixes normal mode and breaks the visual one 😔. I'll try to have another go at it.

Reply:
I have added a mode dependent correction factor. It feels very hacky but it does seem to fix the problem.